src/main/java/gregtech/integration/groovy/GroovyMaterialBuilderExpansion.java

```diff
@@ -85,22 +89,29 @@ public static Material.Builder blastTemp(Material.Builder builder, int temp, Str
 
     public static Material.Builder blastTemp(Material.Builder builder, int temp, String raw, int eutOverride,
                                              int durationOverride, int vacuumEUtOverride, int vacuumDurationOverride) {
-        BlastProperty.GasTier gasTier = null;
-        String name = raw.toUpperCase();
-        for (BlastProperty.GasTier gasTier1 : BlastProperty.GasTier.VALUES) {
-            if (gasTier1.name().equals(name)) {
-                gasTier = gasTier1;
-                break;
-            }
-        }
-        final BlastProperty.GasTier finalGasTier = gasTier;
-        if (GroovyScriptModule.validateNonNull(gasTier, () -> "Can't find gas tier for " + name +
-                " in material builder. Valid values are 'low', 'mid', 'high', 'higher', 'highest'!")) {
+        BlastProperty.GasTier gasTier = GroovyScriptModule.parseAndValidateEnumValue(BlastProperty.GasTier.class, raw,
+                "gas tier");
+        if (gasTier != null) {
             return builder.blast(b -> b
-                    .temp(temp, finalGasTier)
+                    .temp(temp, gasTier)
                     .blastStats(eutOverride, durationOverride)
                     .vacuumStats(vacuumEUtOverride, vacuumDurationOverride));
         }
         return builder;
     }
+
+    public static Material.Builder components(Material.Builder builder, Object... objects) {
+        ObjectArrayList<MaterialStack> materialStacks = new ObjectArrayList<>();
+        for (Object o : objects) {
+            if (o instanceof MaterialStack materialStack) {
+                materialStacks.add(materialStack);
+            } else if (o instanceof Material material) {
+                materialStacks.add(new MaterialStack(material, 1));
+            } else {
+                GroovyLog.get()
+                        .error("Material components must be of type Material or MaterialStack, but was of type {}");
```

Comment:
did you forget to pass in something for formatting the logged message?

Reply:
yes